core/domain/user_jobs_one_off.py

```diff
@@ -468,6 +468,172 @@ def reduce(key, values):
         yield (key, len(values))
 
 
+class FixUserSettingsCreatedOnOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that fixes the invalid values of created_on attribute in the
+    UserSettingsModel. It is a one-off job and can be removed from the codebase
+    after we resolve this issue by running the job once.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(FixUserSettingsCreatedOnOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_settings_model):
+        user_id = user_settings_model.id
+
+        user_dates_list = [
+            user_settings_model.created_on,
+            user_settings_model.last_updated,
+            user_settings_model.last_agreed_to_terms,
+            user_settings_model.last_started_state_editor_tutorial,
+            user_settings_model.last_started_state_translation_tutorial,
+            user_settings_model.last_logged_in,
+            user_settings_model.last_edited_an_exploration,
+            user_settings_model.last_created_an_exploration,
+            user_settings_model.first_contribution_msec
+        ]
+
+        model_names = ['UserSettingsModel'] * 9
```

Comment:
I think it would be better to do something like

```
 user_dates_list = [
    ('UserSettingsModel', user_settings_model.created_on),
    ('UserSettingsModel', user_settings_model.last_updated),
    ('UserSettingsModel', user_settings_model.last_agreed_to_terms),
    …
]
```

and then down below do `model_name, min_date = min(filtered_user_dates_list, key=lambda x: x[1])`, that way you can have just one list with tuples instead of keeping two lists.

Reply:
Are you suggesting to eliminate `models_linked_with_user_settings_model` by this? I don't get how this will work.
This approach is good for `user_settings_model` since we know that it exists, but the same is not true for other models. I have to check for other models, and only when it exists for the user, I can append its date-time attribute to the list. That is why I thought of keeping models and their names as a list of tuples so that I can loop over it, instead of having to repeatedly write similar code for checking each model (i.e to avoid writing `if model is not None` condition after fetching each model). Thoughts @vojtechjelinek?

Reply:
What you can do instead of
```
for model_name, model in models_linked_with_user_settings_model:
    if model is not None:
        user_dates_list.append(model.last_updated)
        user_dates_list.append(model.created_on)
        model_names.append(model_name)
        model_names.append(model_name)
```
is something like 
```
for model_name, model in models_linked_with_user_settings_model:
    if model is not None:
        user_dates_list.append((model_name, model.last_updated))
        user_dates_list.append((model_name, model.created_on))
```
rigth?

Reply:
So there would be no need for any additional code I guess.

Reply:
My bad, I didn't get that earlier. Thanks for the explanation. Did the change now.